docs/reference/modules/discovery.asciidoc

```diff
@@ -1,30 +1,805 @@
+GET /_cluster/state?filter_path=metadata.cluster_coordination.last_committed_config
+--------------------------------------------------
+// CONSOLE
+
+NOTE: The current voting configuration is not necessarily the same as the set of
+all available master-eligible nodes in the cluster. Altering the voting
+configuration itself involves taking a vote, so it takes some time to adjust the
+configuration as nodes join or leave the cluster. Also, there are situations
+where the most resilient configuration includes unavailable nodes, or does not
+include some available nodes, and in these situations the voting configuration
+will differ from the set of available master-eligible nodes in the cluster.
+
+Larger voting configurations are usually more resilient, so Elasticsearch will
+normally prefer to add master-eligible nodes to the voting configuration once
+they have joined the cluster. Similarly, if a node in the voting configuration
+leaves the cluster and there is another master-eligible node in the cluster that
+is not in the voting configuration then it is preferable to swap these two nodes
+over, leaving the size of the voting configuration unchanged but increasing its
+resilience.
+
+It is not so straightforward to automatically remove nodes from the voting
+configuration after they have left the cluster, and different strategies have
+different benefits and drawbacks, so the right choice depends on how the cluster
+will be used and is controlled by the following setting.
+
+`cluster.auto_shrink_voting_configuration`::
+
+    Defaults to `true`, meaning that the voting configuration will automatically
+    shrink, shedding departed nodes, as long as it still contains at least 3
+    nodes.  If set to `false`, the voting configuration never automatically
+    shrinks; departed nodes must be removed manually using the
+    <<modules-discovery-adding-removing-nodes,voting configuration exclusions API>>.
+
+NOTE: If `cluster.auto_shrink_voting_configuration` is set to `true`, the
+recommended and default setting, and there are at least three master-eligible
+nodes in the cluster, then Elasticsearch remains capable of processing
+cluster-state updates as long as all but one of its master-eligible nodes are
+healthy.
+
+There are situations in which Elasticsearch might tolerate the loss of multiple
+nodes, but this is not guaranteed under all sequences of failures. If this
+setting is set to `false` then departed nodes must be removed from the voting
+configuration manually, using the vote withdrawal API described below, to
+achieve the desired level of resilience.
+
+Note that Elasticsearch will not suffer from a "split-brain" inconsistency
+however it is configured. This setting only affects its availability in the
+event of the failure of some of its nodes, and the administrative tasks that
+must be performed as nodes join and leave the cluster.
+
+[float]
+==== Even numbers of master-eligible nodes
```

Comment:
This feels like a very basic notion. Maybe we should move it higher up in the doc to increase exposure?

Reply:
Again, this is low-level details. Interesting if you want to know all the corners, but knowing this won't affect the behaviour of 99.9% of users, and may confuse some, so I think it's right that it's down here in the "if you're interested" section.